app/cdap/components/AbstractWidget/ToggleSwitchWidget/index.tsx

```diff
@@ -60,8 +62,22 @@ const ToggleSwitchWidgetView: React.FC<IToggleToggleSwitchProps> = ({
   const isOn = value === onValue;
 
   function toggleSwitch() {
-    onChange(isOn ? offValue : onValue);
+    // if connection toggle
+    if (isOn && extraConfig.properties.hasOwnProperty('useConnection')) {
+      updateAllProperties(
```

Comment:
I don't think we want to do this. It's GCP-specific - it adds these fields for all sources, even if they don't use them. And what if we want to set defaults for other types, like AWS, or databases (once we support connections there again)?

It also ties this widget to other fields in a way we've tried to avoid so far.

The only way I can think of to handle this now would be in `ConfigurationGroup` in `handleValueChanges`: if any fields are newly shown, have no value, and have a default, set the default. If that seems like too much work, then we should re-prioritize.